mllib/src/test/scala/org/apache/spark/mllib/linalg/distributed/BlockMatrixSuite.scala

```diff
@@ -304,10 +304,25 @@ class BlockMatrixSuite extends SparkFunSuite with MLlibTestSparkContext {
     val largeB = new BlockMatrix(sc.parallelize(largerBblocks, 8), 4, 4)
     val largeC = largeA.multiply(largeB)
     val localC = largeC.toLocalMatrix()
+    val largeC2 = largeA.multiply(largeB, 2)
```

Comment:
I would rather write a test function `testMultiply` which takes `A`, `B`, and expected `C` and make `numSplits` configurable. Here you can simply have 
```scala
testMultiply(largeA, largeB, largeC, 1)
testMultiply(largeA, largeB, largeC, 2)
testMultiply(largeA, largeB, largeC, 3)
testMultiply(largeA, largeB, largeC, 4)
```